### PR TITLE
R&R: Implement basic sort & search functionality

### PR DESCRIPTION
**Pull request by davidguy3237**
Because of the jank way the API is set up, it only does the sorting after getting the reviews. Reworked my localhost GET request so that it grabs 1000 reviews, effectively getting all of the reviews in order to sort them.

### CODE REVIEW

File: client/src/components/ratings-and-reviews/reviews/ReviewsList.jsx [hunk 1]
```diff
@@ -5,54 +5,65 @@ import ReviewTile from './ReviewTile';
...
         product_id: product.id,
+        sort: e.target.value,
       },
     })
       .then(({ data }) => {
-        setReviews((currentReviews) => [...currentReviews, ...nextPage]);
-        setNextPage(data);
-        setPage(page + 1);
+        setShownReviews(data.slice(0, shownReviews.length));
+        setReviewsCache(data);
```
Comment (beanjermin) [inline client/src/components/ratings-and-reviews/reviews/ReviewsList.jsx:44]: look slike youre missing a couple catches

File: client/src/components/ratings-and-reviews/reviews/ReviewsList.jsx [hunk 2]
```diff
@@ -5,54 +5,65 @@ import ReviewTile from './ReviewTile';
...
-          params: {
-            page: 2,
-            product_id: product.id,
-          },
-        }))
-        .then(({ data }) => setNextPage(data));
+        .then(({ data }) => {
+          setShownReviews(data.slice(0, 2));
+          setReviewsCache(data);
+        });
```
Comment (beanjermin) [inline client/src/components/ratings-and-reviews/reviews/ReviewsList.jsx]: catch

**Review by beanjermin — APPROVED** [review]
just missing a couple catches but looks good to me

**Review by beanjermin — DISMISSED** [review]
just missing a couple catches but looks good to me

**Review by beanjermin — DISMISSED** [review]
looks good to go!

**Comment by davidguy3237** [comment]
I forgot to merge the previous PR...Pushed search functionality as well.

**Review by beanjermin — APPROVED** [review]
looks good to me

**Pull request merged**